lib/ansible/galaxy/api.py

```diff
@@ -896,12 +908,11 @@ def get_collection_versions(self, namespace, name):
             if not next_link:
                 break
             elif relative_link:
-                # TODO: This assumes the pagination result is relative to the root server. Will need to be verified
-                # with someone who knows the AH API.
-
-                # Remove the query string from the versions_url to use the next_link's query
-                versions_url = urljoin(versions_url, urlparse(versions_url).path)
-                next_link = versions_url.replace(versions_url_info.path, next_link)
+                next_link_info = urlparse(next_link)
+                next_link = versions_url_info._replace(
+                    query=next_link_info.query,
+                    path=urljoin(versions_url_info.path, next_link_info.path)
+                ).geturl()
```

Comment:
I think this could just be `next_link = urljoin(self.api_server, next_link)` for the scenarios ansible-galaxy expects to handle.

I'm not able to reproduce https://github.com/ansible/ansible/issues/84765 with AH, only with a custom server by making it return a relative path in links. We don't have href in the response from the versions endpoint, so is it safe to assume relative paths are relative to the requested URL rather than a potential redirect? Wondering if we need this here too:
```suggestion
                if not next_link_info.scheme and not next_link_info.path.startswith('/'):
                    raise AnsibleError(f'Invalid non absolute pagination link: {next_link}')
                next_link = urljoin(self.api_server, next_link)
```

Reply:
Hm, I suppose you are right, my implementation was inconsistent, and I was attempting to support something in one place, when I said it couldn't be supported elsewhere.